backend/eventloop/application_event_loop/metrics_test.go

```diff
@@ -0,0 +1,147 @@
+				WithScheme(scheme).
+				WithObjects(gitopsDepl, workspace, argocdNamespace, kubesystemNamespace).
+				Build()
+
+			k8sClient = &sharedutil.ProxyClient{
+				InnerClient: k8sClientOuter,
+				Informer:    &informer,
+			}
+
+			dbQueries, err = db.NewUnsafePostgresDBQueries(false, false)
+			Expect(err).To(BeNil())
+
+			appEventLoopRunnerAction = applicationEventLoopRunner_Action{
+				getK8sClientForGitOpsEngineInstance: func(gitopsEngineInstance *db.GitopsEngineInstance) (client.Client, error) {
+					return k8sClient, nil
+				},
+				eventResourceName:           gitopsDepl.Name,
+				eventResourceNamespace:      gitopsDepl.Namespace,
+				workspaceClient:             k8sClient,
+				log:                         log.FromContext(context.Background()),
+				sharedResourceEventLoop:     shared_resource_loop.NewSharedResourceLoop(),
+				workspaceID:                 workspaceID,
+				testOnlySkipCreateOperation: true,
+			}
+		})
+
+		It("Should update existing deployment, instead of creating new.", func() {
+
+			numberOfGitOpsDeploymentsInErrorState := testutil.ToFloat64(metrics.GitopsdeplFailures)
+			totalNumberOfGitOpsDeploymentMetrics := testutil.ToFloat64(metrics.Gitopsdepl)
+
+			By("passing the invalid GitOpsDeployment into application event reconciler, and expecting an error")
+
+			eventLoopEvent := eventlooptypes.EventLoopEvent{
+				EventType: eventlooptypes.DeploymentModified,
+				Request: reconcile.Request{NamespacedName: types.NamespacedName{
+					Namespace: gitopsDepl.Namespace,
+					Name:      gitopsDepl.Name,
+				}},
+				Client:                  k8sClient,
+				ReqResource:             eventlooptypes.GitOpsDeploymentTypeName,
+				AssociatedGitopsDeplUID: string(gitopsDepl.UID),
+				WorkspaceID:             workspaceID,
+			}
+
+			shutdownSignalled, err := handleDeploymentModified(ctx, &eventLoopEvent, appEventLoopRunnerAction, dbQueries, log.FromContext(context.Background()))
+			Expect(shutdownSignalled).To(BeFalse())
+
+			Expect(err).ToNot(BeNil())
+			newTotalNumberOfGitOpsDeploymentMetrics := testutil.ToFloat64(metrics.Gitopsdepl)
+			newNumberOfGitOpsDeploymentsInErrorState := testutil.ToFloat64(metrics.GitopsdeplFailures)
+			Expect(newTotalNumberOfGitOpsDeploymentMetrics).To(Equal(totalNumberOfGitOpsDeploymentMetrics + 1))
+			Expect(newNumberOfGitOpsDeploymentsInErrorState).To(Equal(numberOfGitOpsDeploymentsInErrorState + 1))
+
+			By("deleting the invalid GitOpsDeployment and calling deploymentModified again")
+
+			err = k8sClient.Delete(ctx, gitopsDepl)
+			Expect(err).To(BeNil())
+			shutdownSignalled, err = handleDeploymentModified(ctx, &eventLoopEvent, appEventLoopRunnerAction, dbQueries, log.FromContext(context.Background()))
+			Expect(shutdownSignalled).To(BeFalse())
+			Expect(err).To(BeNil())
+
+			newNumberOfGitOpsDeploymentsInErrorState = testutil.ToFloat64(metrics.GitopsdeplFailures)
+			Expect(newNumberOfGitOpsDeploymentsInErrorState).To(Equal(numberOfGitOpsDeploymentsInErrorState))
```

Comment:
Can we also check here that the total in non-error state went back down to 0? E.g.
```go
			newNumberOfGitOpsDeploymentsInErrorState = testutil.ToFloat64(metrics.GitopsdeplFailures)
			newTotalNumberOfGitOpsDeploymentMetrics = testutil.ToFloat64(metrics.Gitopsdepl)
			Expect(newNumberOfGitOpsDeploymentsInErrorState).To(Equal(numberOfGitOpsDeploymentsInErrorState))
			Expect(newTotalNumberOfGitOpsDeploymentMetrics).To(Equal(numberOfGitOpsDeploymentsInErrorState))
```